lib/http_ntlm.c

```diff
@@ -141,13 +143,25 @@ CURLcode Curl_output_ntlm(struct connectdata *conn, bool proxy)
     allocuserpwd = &conn->allocptr.proxyuserpwd;
     userp = conn->http_proxy.user;
     passwdp = conn->http_proxy.passwd;
+#if !defined(CURL_DISABLE_CRYPTO_AUTH) || defined(USE_KERBEROS5) || \
+  defined(USE_SPNEGO) || defined(HAVE_GSSAPI)
+	service = conn->data->set.str[STRING_PROXY_SERVICE_NAME] ?
+              conn->data->set.str[STRING_PROXY_SERVICE_NAME] : "HTTP";
+#endif
```

Comment:
turned out the `#if` is necessary because `STRING_PROXY_SERVICE_NAME` is defined under that condition, see urldata.h

Reply:
Oh no, good catch! I always forget about cases like "without crypto auth but with NTLM" :-(

A similar change would also be needed in `curl_sasl.c`, wouldn't it? Then it would probably be easier and more consistent to just enable those two strings for `USE_NTLM`, and also set them in `setopt.c`.

Or just remove the conditions completely? What do you think, @bagder ? The conditions for defining `STRING_PROXY_SERVICE_NAME` and `STRING_SERVICE_NAME` have diverged massively even though they're used together most of the time and even the condition for defining `STRING_SERVICE_NAME` doesn't match the condition for setting it from `CURLOPT_SERVICE_NAME` anymore. The cost would be two additional pointers per easy handle if GSS-API, SSPI, Kerberos, and crypto auth are disabled.

Reply:
Well, I messed up a little.
Now I got compilation error and after closer look it turned out that `STRING_PROXY_SERVICE_NAME` is defined under condition ([urldata.h:1425](https://github.com/curl/curl/blob/67bd4ab19e2903b45b66278956801a681fc35520/lib/urldata.h#L1425))

    #if defined(HAVE_GSSAPI) || defined(USE_WINDOWS_SSPI)
      STRING_PROXY_SERVICE_NAME, /* Proxy service name */
    #endif

and it should have been used here on line 146, and `STRING_SERVICE_NAME` is defined under different condition ([urldata.h:1428](https://github.com/curl/curl/blob/67bd4ab19e2903b45b66278956801a681fc35520/lib/urldata.h#L1428))

    #if !defined(CURL_DISABLE_CRYPTO_AUTH) || defined(USE_KERBEROS5) || \
      defined(USE_SPNEGO) || defined(HAVE_GSSAPI)
      STRING_SERVICE_NAME,    /* Service name */
    #endif

which I used for the first one.
I will fix it in a moment.
Also if you decide to drop those conditions altogether it will help.